jetty-websocket/websocket-servlet/src/main/java/org/eclipse/jetty/websocket/servlet/ServletUpgradeRequest.java

```diff
@@ -314,6 +315,14 @@ public URI getRequestURI()
         return requestURI;
     }
 
+    /**
+     * @return the path within the context, combination of the ServletPath with the PathInfo.
+     */
+    public String getPathInContext()
+    {
+        return URIUtil.addPaths(request.getServletPath(), request.getPathInfo());
```

Comment:
This looks like it may be called a few times per upgade? if so, perhaps do the addPaths in the constructor.

Reply:
It is only called once per upgrade, when creating the FrameHandler.